### PR TITLE
SIRI-724: improve logging for jsoncall

### PR DESCRIPTION
**Pull request by qw3ry**
~~Needs https://github.com/scireum/sirius-kernel/pull/438~~

### CODE REVIEW

File: src/main/java/sirius/web/services/JSONCall.java [hunk 1]
```diff
@@ -60,14 +65,29 @@ public static JSONCall to(URI url, String contentType) throws IOException {
         return result;
     }
 
+    /**
+     * Log the outcall to {@code logger}.
```
Comment (sabieber) [inline src/main/java/sirius/web/services/JSONCall.java:69]:
```suggestion
     * Logs the outcall to {@code logger}.
```

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/web/services/JSONCall.java [hunk 2]
```diff
@@ -60,14 +65,29 @@ public static JSONCall to(URI url, String contentType) throws IOException {
...
+        this.logger = logger;
+        return this;
+    }
+
     /**
      * Adds a custom header field to the call
      *
      * @param name  name of the field
      * @param value value of the field
+     * @return returns the JSON call itself for fluent method calls
```
Comment (jakobvogel) [inline src/main/java/sirius/web/services/JSONCall.java:86]:
```suggestion
     * @return the JSON call itself for fluent method calls
```

**Review by jakobvogel — APPROVED** [review]

File: src/main/java/sirius/web/services/JSONCall.java [hunk 3]
```diff
@@ -82,6 +102,35 @@ public JSONStructuredOutput getOutput() throws IOException {
...
+                                                     """)
+                                     .set("httpMethod", isPost ? "POST" : "GET")
+                                     .set("url", outcall.getRequest().uri())
+                                     .set("callBody", isPost ? getOutput() : null)
+                                     .set("responseCode", getOutcall().getResponseCode())
+                                     .set("response", text)
+                                     .smartFormat());
+                return new ByteArrayInputStream(text.getBytes(StandardCharsets.UTF_8));
+            }
+        }
```
Comment (andyHa) [inline src/main/java/sirius/web/services/JSONCall.java:130]: extract method & ensure that the call isn't executed twice....

**Review by andyHa — CHANGES_REQUESTED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Review by idlira — APPROVED** [review]

File: src/main/java/sirius/web/services/JSONCall.java [hunk 4]
```diff
@@ -82,18 +100,43 @@ public JSONStructuredOutput getOutput() throws IOException {
...
     /**
      * Provides access to the JSON answer of the call.
      *
      * @return the JSON result of the call
      * @throws IOException in case of an IO error while receiving the result
      */
     public JSONObject getInput() throws IOException {
+        String body =
+                Streams.readToString(new InputStreamReader(outcall.getResponse().body(), outcall.getContentEncoding()));
+        logRequest(body);
```
Comment (andyHa) [inline src/main/java/sirius/web/services/JSONCall.java:135]:
```suggestion
        logRequest(body);
        
```

**Review by andyHa — APPROVED** [review]

**Review by mko-sci — APPROVED** [review]

**Pull request merged**